app/controllers/transactions_controller.rb

```diff
@@ -15,21 +15,39 @@ def new
 
   def create
     @transaction = current_user.transactions.new(transaction_params)
-    if @transaction.save
-      redirect_to root_path
-    else
-      render 'new'
+    respond_to do |format|
+      if @transaction.save
+        flash[:notice] = t('transaction_create')
+      else
+        flash[:alert] = t('transaction_not_create')
+      end
```

Comment:
@marderer why have you moved 
```ruby
if @transaction.save
  flash[:notice] = t('transaction_create')
else
  flash[:alert] = t('transaction_not_create')
end
```
block to `respond_to`? What do you think about moving it out of it here and in other places too?